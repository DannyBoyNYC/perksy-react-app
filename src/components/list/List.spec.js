const React = require('react');

const List = require('./index').default;

describe('<List.* />', () => {
  let component;
  const requiredProps = {
    items: ['foo', 'bar']
  };

  describe('<List.Unordered />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <List.Unordered {...requiredProps} />,
        'List'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a <ul> tag', () => {
      expect(component).toMatchSelector('ul');
    });

    it('should apply the onPrerender method to each item if one is given', () => {
      const mockPrerender = jest.fn();
      component = shallowUntilTarget(
        <List.Unordered {...requiredProps} onPrerender={mockPrerender} />,
        'List'
      );
      expect(mockPrerender.mock.calls).toHaveLength(2);
      expect(mockPrerender.mock.calls[0][0]).toBe('foo');
      expect(mockPrerender.mock.calls[1][0]).toBe('bar');
    });
  });

  describe('<List.Ordered />', () => {
    beforeEach(() => {
      component = shallowUntilTarget(
        <List.Ordered {...requiredProps} />,
        'List'
      );
    });

    it('should render correctly with all required props', () => {
      expect(component).toMatchSnapshot();
    });

    it('should render a <ol> tag', () => {
      expect(component).toMatchSelector('ol');
    });
  });
});
