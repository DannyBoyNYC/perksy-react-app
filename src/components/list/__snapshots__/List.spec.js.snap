// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List.* /> <List.Ordered /> should render correctly with all required props 1`] = `
<ol
  className="List-root-0-1-1"
>
  <li
    className="List-item-0-1-2"
    key="item-0"
  >
    foo
  </li>
  <li
    className="List-item-0-1-2"
    key="item-1"
  >
    bar
  </li>
</ol>
`;

exports[`<List.* /> <List.Unordered /> should render correctly with all required props 1`] = `
<ul
  className="List-root-0-1-1"
>
  <li
    className="List-item-0-1-2"
    key="item-0"
  >
    foo
  </li>
  <li
    className="List-item-0-1-2"
    key="item-1"
  >
    bar
  </li>
</ul>
`;
